service/server.go

```diff
@@ -544,6 +552,29 @@ func (s *Server) getNode(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func checkDel(s *Server, user *auth.User, id *uuid.UUID, u *url.URL) (del bool, err error) {
+	delete := false
+	if _, ok := u.Query()["del"]; ok {
+		delete = true
+		// If we want to move the deletion into the core logic it could go like this:
+		// * pass in deletion flag and a pre-write function that accepts a node
+		// * do the ACL checks, etc.
+		// * the pre-write function is called, which writes the headers to the writer
+		//   * the core logic should not know the writer is a http.ResponseWriter
+		// * stream the file
+		// * delete the node
+		// not worth the trouble for now
+		node, err := s.store.Get(user, *id)
+		if err != nil {
+			return false, err
+		}
+		if user == nil || (user.GetUserName() != node.Owner.AccountName && !user.IsAdmin()) {
```

Comment:
L571, if any of condition in () in `False`, then the error will not be raised. This does not sounds right, unless I am missing something.

Reply:
The parentheses say "If the user is not the node owner and the user is not an admin, raise an error", which is the behavior we want

Reply:
I see. So either be the node owner or has admin right is good

Reply:
Right. You could do
```
if user == node.owner or user.isAdmin() {
    //pass
} else {
    throw NewError(...)
```
which would have equivalent logic, but that's just wasted space

Reply:
Is less restricted that I thought. But it makes sense.